src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -194,4 +242,47 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
   set interactorTypeFilter(value: string[]) {
     this._interactorTypeFilter = value;
   }
+
+  public get allInteractorsInComplexSearch(): Interactor[] {
+    return this._allInteractorsInComplexSearch;
+  }
+
+  setListView() {
+    this.DisplayType = this.LIST_VIEW;
+    this.toast = this.notificationService.complexNavigatorAnnouncement();
+    this.reloadPage();
+  }
+
+  setComplexNavigatorView() {
+    this.DisplayType = this.COMPLEX_NAVIGATOR_VIEW;
+    this.reloadPage();
+    if (!!this.toast) {
+      this.notificationService.closeAnnouncement(this.toast.toastId);
+      this.toast = null;
+    }
+  }
+
+
+  setFirstDisplayType(): void {
```

Comment:
To fix the redirect, what we need to do is to only redirect when people do a search. When people do a search, we don't have the display type set up, then we can redirect. When people change a filter, or change pages, or change from list to navigator or the other way around, display type is set, and we don't need to redirect.

So, that means, moving the redirect logic to inside the if `if (!this.DisplayType) {`

Basically, this method ends up like this
```
  setFirstDisplayType(): void {
    if (!this.DisplayType) {
      if (this._complexSearch.totalNumberOfResults === 1) {
        const complexId = this._complexSearch.elements[0].complexAC;
        if (!!complexId) {
          // For some reason this is needed so the navigate call works
          this.router.routeReuseStrategy.shouldReuseRoute = function () {
            return false;
          };
          this.router.navigate(['/complex', complexId]);
        }
      } else {
        // Currently the list view is the default, as we are just launching the navigator view
        // Later on we can change the default view to be the list or navigator view based on number of results
        if (this._complexSearch.totalNumberOfResults <= this._navigatorPageSize) {
          this.setComplexNavigatorView();
        } else {
          this.setListView();
        }
      }
    }
  }
```